common/src/main/java/com/emc/mongoose/common/concurrent/DaemonBase.java

```diff
@@ -12,6 +12,7 @@
 implements Daemon {
 
 	private AtomicReference<State> stateRef = new AtomicReference<>(State.INITIAL);
+	protected final Object state = new Object();
 
 	private enum State {
 		INITIAL, STARTED, SHUTDOWN, INTERRUPTED, CLOSED
```

Comment:
Emm, maybe you don't understand me again. I mean why enum State is not part of Daemon interface. It is not about Object at all.